farge

```diff
@@ -62,29 +64,29 @@ hex_to_rgb() {
 check_dependencies() {
     # Only check colorpicker if it's running in X,else check for slurp and grim
     if [ "$WAYLAND_DISPLAY" ]; then
-        CHECK_SLURP=$(command -v slurp)
-        CHECK_GRIM=$(command -v grim)
-        CHECK_CB=$(command -v wl-copy)
-        [ ${#CHECK_CB} -eq 0 ] && echo "wl-copy needs to be installed: https://github.com/bugaevc/wl-clipboard" && exit 1
-        [ ${#CHECK_SLURP} -eq 0 ] && echo "slurp needs to be installed: https://wayland.emersion.fr/slurp/" && exit 1
-        [ ${#CHECK_GRIM} -eq 0 ] && echo "grim need to be installed: https://wayland.emersion.fr/grim/" && exit 1
+        ! command -v grim &>/dev/null &&
+            echo "grim need to be installed: https://wayland.emersion.fr/grim" && v=" "
+        ! command -v slurp &>/dev/null &&
+            echo "slurp needs to be installed: https://wayland.emersion.fr/slurp" && v=" "
+        ! command -v wl-copy &>/dev/null &&
+            echo "wl-copy needs to be installed: https://github.com/bugaevc/wl-clipboard" && v=" "
+        [ "$v" ] && exit 1
+        unset v
```

Comment:
The variables `$CHECK_<x>` were useless as they were only used in one place, I removed them and spaced it out a bit so the lines aren't 120 chars long. I also set the `$v` variable so it prints all the missing utilities then exits, instead of printing one then exiting.

Reply:
Could you explain why you are using the exclaimation mark? I did a rough understanding from reading [this](https://stackoverflow.com/questions/48748030/in-bash-what-does-exclamation-mark-before-command-means) StackOverflow post, but Im not enierrly sure why it is being used here.

I also dont understand what  ` unset v` does. From my understanding it removes the variable? Why is that needed?

Reply:
Think of `!` as boolean not, it changes non-zero exit code to zero and zero exit-code to non-zero.

There are no local variables in sh, you can make a hacky implementation by using unset-ing a variable. Why would you do that? To keep the namespace clean and to avoid accidental override of variables.

Reply:
@vizs Thank you, that makes sense :+1:

Reply:
@viz

>  zero exit-code to non-zero. FWIW, you can rewrite ! command -v cmd &>/dev/null && echo msg to command -v cmd &>/dev/null || echo msg.

I couldn't find a way of doing this *and* setting the variable.

> There are no local variables in sh, you can make a hacky implementation by
unset-ing a variable. Why would you do that? To keep the namespace
clean and to avoid accidental override of variables.

Come to think of it, I probably should have put `unset` up top, in case the user has `$v` exported in their normal environment for whatever reason.